tests/base_scraper/test_base_scraper.py

```diff
@@ -74,5 +74,39 @@ def test_notimplementedexceptions(self):
         self.assertRaises(mozdownload.NotImplementedError,
                           scraper.build_filename, 'invalid binary')
 
+    def test_authentication(self):
+        """testing with basic authentication"""
+        username = 'mozilla'
+        password = 'mozilla'
+        basic_auth_url = 'http://mozqa.com/data/mozqa.com/http_auth/basic/'
+
+        # test with invalid authentication
+        scraper = mozdownload.DirectScraper(directory=self.temp_dir,
+                                            url = basic_auth_url,
+                                            version=None)
+        self.assertRaises(requests.exceptions.HTTPError,scraper.download)
+
+        # testing with valid authentication
+        scraper = mozdownload.DirectScraper(directory=self.temp_dir,
+                                            url = basic_auth_url,
+                                            version=None,
+                                            username=username,
+                                            password=password)
```

Comment:
I would suggest you also add two more tests when only the username or the password has been given.